src/daemon/screenlock.vala

```diff
@@ -0,0 +1,375 @@
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+namespace Budgie {
+    const string DBUS_SCREENLOCK = "org.buddiesofbudgie.BudgieScreenlock";
+	const string DBUS_SCREENLOCK_PATH = "/org/buddiesofbudgie/Screenlock";
+
+    const string POWERSCREEN_DBUS_NAME = "org.gnome.SettingsDaemon.Power";
+    const string POWERSCREEN_DBUS_OBJECT_PATH = "/org/gnome/SettingsDaemon/Power";
+
+    [DBus (name="org.gnome.SettingsDaemon.Power.Screen")]
+    public interface PowerScreenRemote : GLib.Object {
+        public abstract int32 Brightness { get; set; }
+    }
+
+	[DBus (name="org.buddiesofbudgie.BudgieScreenlock")]
+    public class Screenlock {
+        static Screenlock? instance;
+
+        // Screen Dim capability
+        PowerScreenRemote? powerscreen_proxy = null;
+        bool isdimmable = false;
+        int32 current_brightness = 0;
+
+        // connections to various schemas used in screenlocking
+        private GLib.Settings power;
+        private GLib.Settings session;
+        private GLib.Settings screensaver;
+        private GLib.Settings lockdown;
+
+        // public functions should check this variable first and exit gracefully where needed
+        bool all_apps = true;
+
+        // remember the locker to be used
+        private string locker = "";
+
+        /** Our upower client */
+        private Up.Client client;
+        private GLib.GenericArray<Up.Device> devices;
+        private bool battery_mode;
+
+        private string calculate_lockcommand() {
+            string output = "";
+
+            if (locker == "swaylock") {
+                // grab the background picture and strip the file prefix
+                string current_wallpaper = screensaver.get_string("picture-uri").replace("file://","");
+
+                // swaylock interprets : as a delimiter for the output name
+                // recommendation by swaylock is to replace : with ::
+                current_wallpaper = current_wallpaper.replace(":", "::");
+
+                File file = File.new_for_path(current_wallpaper);
+                if (current_wallpaper == "" || !file.query_exists()) {
+                    output = "-c 000000";
+                }
+                else {
+                    output = "-i " + current_wallpaper;
+                }
+
+                output = "swaylock -Fklf " + output;
+            }
+
+            if (locker == "gtklock") {
+                output = "gtklock -d";
+
+                /**
+                * Try in order, and load the first one that exists:
+                * - /etc/budgie-desktop/[gtklock.ini | gtklock.css]
+                * - /usr/share/budgie-desktop/[gtklock.ini | gtklock.css]
+                */
+                string[] system_configs = {
+                    @"file://$(Budgie.CONFDIR)/budgie-desktop/gtklock.ini",
+                    @"file://$(Budgie.DATADIR)/budgie-desktop/gtklock.ini"
+                };
```

Comment:
Should we care about files in `~/.config`?

Reply:
Fair enough - added in the latest commit